python/cudf/cudf/_lib/column.pyx

```diff
@@ -331,12 +354,8 @@ cdef class Column:
 
         if col.base_data is None:
             data = NULL
-        elif isinstance(col.base_data, SpillableBuffer):
-            data = <void*><uintptr_t>(col.base_data).get_ptr(
-                spill_lock=get_spill_lock()
-            )
         else:
-            data = <void*><uintptr_t>(col.base_data.ptr)
+            data = <void*><uintptr_t>(col.base_data.mutable_ptr)
```

Comment:
> But then what happens if `SpillableBuffer.ptr` becomes something more than a trivial `return self._ptr`?

Today it's non-trivial, we un-spill when `SpillableBuffer.ptr` is accessed and marked as exposed. That is why we have a special casing at line no 408.

Reply:
I don't have opinions on getting rid of `*ptr` properties, but I suspect it is also a huge breaking change across libraries and not in the scope of this PR. However, a utility something like `get_ptr` that returns a `ptr` doing different things for different buffers is something I'm okay to add if you see a necessity to do so.

Reply:
Agreed, I'm inclined to the idea of `get_ptr(access_mode=READ|READ_WRITE|WRITE)`. But let's punt that to a future PR.

Reply:
I agree, let’s do this in a separate PR but I think we should merge such a PR before this one. It will simplify both this PR and the spilling code. E.g., we can replace line `407-416` with:
```python
data = <void*><uintptr_t>(col.base_data.get_ptr(access_mode=READ))
```